enhancements/single-node/single-node-openshift-with-workers.md

```diff
@@ -0,0 +1,495 @@
+installer sets the Scheduler CR `.spec.mastersSchedulable` to `true` and as a
+result the control-plane node is in both the "master" and "worker" pools).
+
+This pod floating between the nodes is a problem because user ingress traffic
+has to be directed at the node currently holding the `router-default` pods, and
+since the DNS entries have been "naively" pointed at the original control-plane
+node's IP address (which may no longer be running those pods), ingress traffic
+can no longer work. This can be temporarily solved if DNS is adjusted to point
+at the correct node currently holding the pod or a load-balancer / some virtual
+IP solution is put in place and then the DNS entry can be directed at that
+load-balancer / virtual IP instead of at the node. This enhancement will try
+to prevent this floating altogether so these workarounds will not be needed.
+
+Finally, performing bootstrap-in-place installation of a "none"-platform single
+control-plane node cluster today using the OpenShift Installer does not provide
+the user with a worker node ignition file. As a result, the user is not able to
+manually add worker nodes to the resulting cluster. All current official OCP
+documentation points the user to use "the worker ignition file that get
+generated during installation", but that is not currently generated in such
+bootstrap-in-place installation.
+
+### Goals
+
+- Deal with the "floating ingress" issue encountered in "none"-platform
+single control-plane node clusters which have worker nodes added to them.
+
+- Define how the installer may be modified to generate the worker ignition
+manifest, even when doing bootstrap-in-place "none"-platform single
+control-plane node cluster installation
+
+### Non-Goals
+
+- Deal with the scalability of single control-plane node clusters that have
+additional workers added to them. The absence of multiple control plane nodes
+means that the number of workers/pods that can be supported on such topology is
+even more limited than in a regular 3 control-plane node cluster. Relevant
+documentation may have to point out that the pod limits (or other control-plane
+related limits) that apply to a single control-plane node cluster with no
+workers also apply globally across all additional added workers. If a large
+amount of workers/pods is desired the user would have to re-install the cluster
+as a regular multi control-plane node cluster.
+
+- Deal with clusters that have less nodes than they had during day-1
+installation. The rationale for this being a non-goal is that it seems that
+removing the "initial" nodes of a cluster causes a lot of issues even in a
+regular, multi-node installation. So dealing with similar issues in single
+control-plane node clusters is also out of scope for this enhancement. For
+example, installing a cluster with 3 control-plane nodes and 3 workers results
+in unschedulable control-plane nodes. Scaling down the worker machine-sets from
```

Comment:
Your understanding is correct. And I understand the confusion from using the blanket term "unschedulable", however I'm just borrowing the terminology from the Scehduler CR entry that controls this behavior:
https://github.com/openshift/api/blob/8e88d13169d91aa9c64c8b6ed8bba0caf4a40d3a/config/v1/0000_10_config-operator_01_scheduler.crd.yaml#L42-L44

Reply:
Yeah, the `mastersSchedulable` field name and documentation suffer from using such a fraught term.  It's probably too much trouble to change the field name now, and fixing the godoc or product documentation is way out of scope for this enhancement.  However, in my opinion, going forward, the terms "schedulable" and "unschedulable" should be qualified or avoided entirely because their various historical uses render the semantics murky and misleading.  That's just my opinion though.  ¯\\\_(ツ)\_/¯